src/auditwheel/tools.py

```diff
@@ -69,15 +113,16 @@ def dir2zip(in_dir: str, zip_fname: str, date_time: datetime | None = None) -> N
     date_time : Optional[datetime]
         Time stamp to set on each file in the archive
     """
+    in_dir = os.path.normpath(in_dir)
     if date_time is None:
         st = os.stat(in_dir)
         date_time = datetime.fromtimestamp(st.st_mtime, tz=timezone.utc)
     date_time_args = date_time.timetuple()[:6]
     compression = zipfile.ZIP_DEFLATED
     with zipfile.ZipFile(zip_fname, "w", compression=compression) as z:
-        for root, dirs, files in os.walk(in_dir):
-            for dir in dirs:
-                dname = os.path.join(root, dir)
+        for root, dirs, files in walk(in_dir):
+            if root != in_dir:
```

Comment:
The following trivial change would skip adding zip file entries for non-empty directories. Empty directories are still added, as that may be important for the case of special wheels with data files packaging core non-Python libraries and executables (e.g. a wheel with a tool  that expects a `$venv_prefix/etc/<toolname>` to exist).

```suggestion
            if root != in_dir and not (dirs or files):
```